CorePluginFramework/src/au/gov/asd/tac/constellation/plugins/parameters/RecentParameterValues.java

```diff
@@ -108,7 +109,8 @@ public static void removeListener(RecentValuesListener listener) {
     public static void fireChangeEvent(RecentValuesChangeEvent e) {
         synchronized (LISTENERS) {
             for (RecentValuesListener listener : LISTENERS) {
-                listener.recentValuesChanged(e);
+                //JavaFX code allows updating the UI from an JavaFX application thread.            
+                Platform.runLater(() -> listener.recentValuesChanged(e));
             }
```

Comment:
Given this is the only code in the for loop, I'm unsure whether it is better to have the `Platform.runLater` call inside the loop or outside?

Reply:
@antares1470 what do you mean? The existing logic was already making the call for each listener. So shouldn't we just continue doing this but @spica198 is just making sure it is run on the JavaFX thread?

Reply:
@arcturus2 what I meant was comparing this solution to queuing the actual for loop itself, given everything in the for loop in getting queued on the FX thread. i.e.
```
Platform.runLater(() -> {
  for (listener : LISTENERS) {
  ...
});
```

Reply:
Both options fix the issue here, but not sure which one is best. Kind of agree with @antares1470. Otherwise it's  overwhelming the event queue with little runnables, most of them would just return in ValueInputPane.recentValuesChanged as the first if condition fails.

Reply:
Agree with @antares1470 .. Covering the loop within thread make sense instead of putting each listener in thread.